app/helpers/insured/group_selection_helper.rb

```
@@ -12,6 +12,10 @@ def can_shop_both_markets?(person)
       can_shop_individual?(person) && can_shop_shop?(person)
     end
 
+    def can_shop_resident?(person)
+      person.try(:has_active_resident_role?)
```

Have you considered using `unless person.blank?` instead of using `.try`? Then throw an error if you haven't fed a person to the method instead of simply returning a nil.